hack/verify-code-patterns.sh

```diff
@@ -43,3 +43,17 @@ if [[ -n "$out" ]]; then
   echo >&2 "$out"
   exit 1
 fi
+
+# Do not initialize index.{Plugin,Platform} structs in test code.
+#
+# TODO(ahmetb): ideally this plugin should cleanly scan all (index\.)(Plugin|Platform) occurrences except slice literals
+# (which are prefixed with []). However I could not get negative lookbehind pattern to work with grep to match to ones
+# that do not have a [] prefix.
+out="$(grep --include '*_test.go' --exclude-dir 'vendor/' -EIrn '\s+(index\.)(Plugin|Platform){' || true)"
+if [[ -n "$out" ]]; then
```

Comment:
also a bit off-topic to this PR (and hacky). but seems to be doing the job so far. ;)

Reply:
I think it's a good idea to have this test. If we find it is too restricting, we can still re-evaluate again.